pyrigi/_flexible_graph_stable_cut.py

```
@@ -0,0 +1,322 @@
+"""
+Algorithm for a stable cut search in a flexible graph
+according to Algorithm 1 in :cite:p:`ClinchGaramvölgyiEtAl2024`.
+"""
+
+import logging
+from typing import Optional
+
+
+from pyrigi.data_type import StableCut
+from pyrigi._cuts import stable_set_violation
+import networkx as nx
+import numpy as np
+
+from pyrigi.data_type import Vertex
+
+
+def stable_cutset_in_flexible_graph(
```

Please, rename it to `stable_separating_set` and introduce parameter `check_flexible` instead, similarly to `check_min_rigid` in https://pyrigi.github.io/PyRigi/userguide/api/graph.html#pyrigi.graph.Graph.number_of_realizations

---

I don't think I understand what the expected behavior should be: You mean, an exception will be thrown for flexible graphs instead of the function returning `None` in it's `Optional` return type?

---

I mean to add `check_flexible=True` to the signature and at the beginning of the function::
```py
if check_flexible and graph.is_rigid(dim=2):
    raise ValueError("The graph must be 2-flexible!")
```